packages/terminal/src/browser/terminal-quick-open-service.ts

```diff
@@ -0,0 +1,124 @@
+
+@injectable()
+export class TerminalQuickOpenService implements QuickOpenModel, QuickOpenHandler {
+
+    @inject(PrefixQuickOpenService)
+    protected readonly prefixQuickOpenService: PrefixQuickOpenService;
+
+    @inject(WidgetManager)
+    protected readonly widgetManager: WidgetManager;
+
+    @inject(ApplicationShell)
+    protected readonly shell: ApplicationShell;
+
+    readonly prefix: string = 'term ';
+
+    get description(): string {
+        return 'Show All Opened Terminals';
+    }
+
+    getModel(): QuickOpenModel {
+        return this;
+    }
+
+    getOptions(): QuickOpenOptions {
+        return {
+            fuzzyMatchLabel: {
+                enableSeparateSubstringMatching: true
+            },
+            fuzzyMatchDescription: {
+                enableSeparateSubstringMatching: true
+            }
+        };
+    }
+
+    open(): void {
+        this.prefixQuickOpenService.open(this.prefix);
+    }
+
+    get all(): TerminalWidget[] {
+        return this.widgetManager.getWidgets(TERMINAL_WIDGET_FACTORY_ID) as TerminalWidget[];
+    }
+
+    async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        const terminalItems: QuickOpenItem[] = [];
+
+        const widgets: TerminalWidget[] = this.all.sort();
+
+        if (widgets.length === 0) {
+            terminalItems.push(new QuickOpenItem({
+                label: 'List of opened terminals is currently empty',
+                run: () => false
+            }));
+            acceptor(terminalItems);
+            return;
+        }
+
+        for (const widget of widgets) {
+            const item = await this.toItem(widget);
+            terminalItems.push(item);
+        }
+        acceptor(terminalItems);
+        return;
+    }
+
+    /**
+     * Converts the terminal widget to the quick open item.
+     * @param {TerminalWidget} widget - the terminal widget.
+     * @returns Promise of the quick open item.
+     */
+    protected async toItem(widget: TerminalWidget): Promise<QuickOpenItem<QuickOpenItemOptions>> {
+
+        const options: QuickOpenItemOptions = {
+            label: widget.id,
+            iconClass: widget.title.iconClass,
+            description: widget.title.label,
+            tooltip: '',
+            hidden: false,
+            run: this.getRunFunction(widget)
+        };
+        return new QuickOpenItem<QuickOpenItemOptions>(options);
+    }
+
+    /**
+     * Gets the function that can open the editor file.
+     * @param {TerminalWidget} widget - the terminal widget to be opened.
+     * @returns Function that would open the terminal if mode === QuickOpenMode.OPEN.
+     */
+    protected getRunFunction(widget: TerminalWidget): (mode: QuickOpenMode) => boolean {
+
+        return (mode: QuickOpenMode) => {
+            if (mode !== QuickOpenMode.OPEN) {
+                return false;
+            }
+            this.shell.activateWidget(widget.id);
```

Comment:
I believe you can simply call `terminalService.open` at this point and drop the injection of `ApplicationShell`.